### PR TITLE
Fix pwnam (HPC-8533)

### PR DESCRIPTION
**Pull request by itkovian**

### CODE REVIEW

File: lib/vsc/filesystem/quota/tools.py [hunk 1]
```diff
@@ -184,6 +184,7 @@ def process_user_quota(storage, gpfs, storage_name, filesystem, quota_map, user_
     """
     del filesystem
     del gpfs
+    del user_map
```
Comment (stdweird) [inline lib/vsc/filesystem/quota/tools.py]: where does this come from? does it work anywhere at all?

or can we populate it and reuse the lookups for all the filesystems?

Comment (itkovian) [reply lib/vsc/filesystem/quota/tools.py]: Hmm, good question. I guess we can do that.

File: lib/vsc/filesystem/quota/tools.py [hunk 2]
```diff
@@ -199,15 +199,23 @@ def process_user_quota(storage, gpfs, storage_name, filesystem, quota_map, user_
...
-            if not user_name:
+            try:
+                user_name = user_map.get(user_id, pwd.getpwuid(int(user_id)).pw_name)
+            except KeyError:
+                logging.debug("Cannot find user ID %s in the user mapping", user_id)
+                continue
+            if not user_name.startswith('vsc'):
                 continue
 
+            if user_id not in user_map:
```
Comment (stdweird) [inline lib/vsc/filesystem/quota/tools.py:210]: move on try block, is also usefull for non-vsc lookups i guess

Comment (itkovian) [reply lib/vsc/filesystem/quota/tools.py:210]: We already continued and skipped the rest of the iteration if the user name doe not start with vsc.

Comment (itkovian) [reply lib/vsc/filesystem/quota/tools.py:210]: ping @stdweird I do not grok what you meant ^

Comment (stdweird) [reply lib/vsc/filesystem/quota/tools.py:210]: adding the resolved userid to the map/cache is also relevant for non-vsc users. so you can this earlier. perhaps not in the try block, but before previous `if` statement

Comment (itkovian) [reply lib/vsc/filesystem/quota/tools.py:210]: I fail to see why. We only store quota for VSC users, right?

Comment (stdweird) [reply lib/vsc/filesystem/quota/tools.py:210]: the cache is only to speed up the lookups, no need to have "slower" lookups for users even if we don't care about the data

**Comment by itkovian** [comment]
Superseded by #15

**Pull request closed without merging**